src/client/auth/aws.rs

```diff
@@ -197,6 +214,119 @@ pub(crate) async fn get_aws_credentials(credential: &Credential) -> Result<Crede
     }
 }
 
+pub async fn compute_aws_sigv4_payload(
```

Comment:
Does this need to be `async`?  I didn't see anything in the body that required it.